olaaf_django/sync_hashes.py

```diff
@@ -87,22 +96,51 @@ def _insert_diff_hashes(repo, prev_commit, current_commit):
       # if file was added or modified, calculate the new hash
       file_type = 'html' if file_name.endswith('.html') else 'pdf'
       if file_type == 'html':
-        url = path.rsplit('.', 1)[0]
+        if 'index.html' in path:
+          url = path.split('index.html')[0]
+          if url:
+            url = url.rsplit('/', 1)[0]
+        else:
+          url = path.rsplit('.', 1)[0]
+        hash_type = Hash.RENDERED
+
       else:
         url = path
+        hash_type = Hash.BITSTREAM
       # if file aready existed and it was modified or deleted update previous hash
       if action != 'A':
-        previous_hash = Hash.objects.get(path=url, end_commit__isnull=True)
-        previous_hash.end_commit = current_commit
-        previous_hash.save()
+        q = Q(path=url, end_commit__isnull=True)
+        if query is None:
+          query = q
+        else:
+          query = query | q
+        end_commits_by_path[url] = current_commit
```

Comment:
what is the purpose of end_commits_by_path? It appears that every key will have the same value: current_commit. Why not just directly set current_commit on line 139?